core/templates/pages/exploration-editor-page/modal-templates/welcome-modal.component.spec.ts

```diff
@@ -0,0 +1,99 @@
+// Copyright 2021 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Unit tests for WelcomeModalComponent.
+ */
+
+import { Component, NO_ERRORS_SCHEMA } from '@angular/core';
+import { ComponentFixture, waitForAsync, TestBed } from '@angular/core/testing';
+import { NgbActiveModal } from '@ng-bootstrap/ng-bootstrap';
+import { UrlInterpolationService } from
+  'domain/utilities/url-interpolation.service';
+import { ContextService } from 'services/context.service';
+import { SiteAnalyticsService } from 'services/site-analytics.service';
+import { importAllAngularServices } from 'tests/unit-test-utils.ajs';
+import { WelcomeModalComponent } from './welcome-modal.component';
+@Component({
+  selector: 'oppia-changes-in-human-readable-form',
+  template: ''
+})
+class ChangesInHumanReadableFormComponentStub {
+}
+
+class MockActiveModal {
+  close(): void {
+    return;
+  }
+
+  dismiss(): void {
+    return;
+  }
+}
+
+describe('Welcome Modal Component', () => {
+  let component: WelcomeModalComponent;
+  let fixture: ComponentFixture<WelcomeModalComponent>;
+  let ngbActiveModal: NgbActiveModal;
+  let contextService: ContextService;
+  let siteAnalyticsService: SiteAnalyticsService;
+  const explorationId = 'exp1';
+  let siteAnalyticsServiceSpy: jasmine.Spy;
+
+  importAllAngularServices();
```

Comment:
This is not required for angular unit tests.

Reply:
I deleted it!